pkg/credentialprovider/provider.go

```diff
@@ -58,9 +58,9 @@ type CachingDockerConfigProvider struct {
 	Provider DockerConfigProvider
 	Lifetime time.Duration
 
-	// ShouldCache is an optional function that returns true if the specific config should be cached.
-	// If nil, all configs are treated as cacheable.
-	ShouldCache func(DockerConfig) bool
+	// ShouldUseCache is an optional function that returns true(by default) if use cache from specific config.
+	// If nil, should also use cache
+	ShouldUseCache func(string, DockerConfig) bool
```

Comment:
name the parameters, the meaning of `string` is not clear

Reply:
fixed and also updated the func `ShouldUseCache` comments